python/plugins/processing/algs/qgis/TilesXYZ.py

```diff
@@ -172,103 +177,132 @@ def prepareAlgorithm(self, parameters, context, feedback):
         self.layers = [l for l in project.layerTreeRoot().layerOrder() if l in visible_layers]
         return True
 
+    def render_single_tile(self, metatile):
+        if self.feedback.isCanceled():
+            return
+            # Haven't found a better way to break than to make all the new threads return instantly
+
+        if "Dummy" in threading.current_thread().name: # single thread testing
+            specific_settings = list(self.settings_dict.values())[0]
+        else:
+            try:
+                # guess we're not the only one using TPE in QGIS, cannot assume 0_#, it's sometimes 3 or 4_#
+                specific_settings = self.settings_dict[threading.current_thread().name[-1]] # last number only
+            except:
+                print("Exception! our threads don't match with our settings! ")
+
+        size = QSize(self.tile_width * metatile.rows(), self.tile_height * metatile.columns())
+        self.extent = QgsRectangle(*metatile.extent())
```

Comment:
self.extent is it shared among all threads?

Reply:
As I understand the code, and I might be wrong, extent here is the total area that defines the outer boundries that we render to, like the canvas in an image manipulation program.

Sharing it between all threads is fine because it doesn't change per job/tile it changes only per rendering job. Can probably be removed  as extraneous since the thread can find the extent without it being saved to the class though...